sphinxgallery/gen_rst.py

```diff
@@ -444,7 +458,12 @@ def generate_file_rst(fname, target_dir, src_dir):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot'):
+    exec_script = fname.startswith('plot')
+    use_mayavi = gallery_conf.get('use_mayavi', False)
+    if not use_mayavi and any('mayavi' in bl[1] for bl in script_blocks):
```

Comment:
Sounds good. Thinking about it, I reckon this is probably cleaner to do everything in conf.py because you want it to work locally for people whether they have mayavi installed or not.

So my proposal is to have something like:

```python
try:
    from mayavi import mlab
    find_mayavi_figures = True
    examples_dir = ['../examples', '../tutorials', '../mayavi_examples']
except ImportError:
    find_mayavi_figures = False
    examples_dir = ['../examples', '../tutorials']
```

Same thing for gallery_dir and use it in sphinx_gallery_conf.

and then use examples_dir and gallery_dir in sphinx_gallery_conf.